ast/index.go

```diff
@@ -13,11 +13,8 @@ type Index struct {
 }
 
 func (n *Index) Accept(v Visitor) Node {
-	// the Key may have further interpolations
-	switch n.Key.(type) {
-	case *Call, *VariableAccess:
```

Comment:
So my understanding is that dropping this conditional check allows keys that are arithmetic nodes to get pushed onto the type checking stack, which then get transformed into call nodes, which then get evaluated? Whereas, previously, index keys that were arithmetic nodes would be let un-type-checked, and could not be handled at the type-checking phase.